src/hooks/use-on-screen.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * hook that check if a component is on view
+ *
+ * @param {Ref} ref component reference
+ * @return {Boolean}  is component in view
+ */
+function useOnScreen(ref) {
+  const [isIntersecting, setIntersecting] = useState(false);
```

Comment:
Please stick to the best practices for function naming; This is pretty essential 
```suggestion
  const [isIntersecting, setIsintersecting] = useState(false);
```